### PR TITLE
i18n(vi): add Vietnames translation for Starlight UI

### PR DESCRIPTION
**Pull request by torn4dom4n**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description

- Add Vietnamese translation for Starlight UI

I'm taking part in Hacktoberfest.

### CODE REVIEW

**Review by HiDeoo — COMMENTED** [review]
Thanks for your contribution :raised_hands:

From a purely technical point of view, adding a new language will require 2 additional changes:

- Adding the language to the list of supported languages in the documentation (https://github.com/withastro/starlight/blob/main/docs/src/content/docs/guides/i18n.mdx#L146)
- Adding a changeset describing the change (you can easily create one by running `pnpm exec changeset`)

If needed, [this PR](https://github.com/withastro/starlight/pull/537/files) adding the Swedish language can be used as a reference.

_I've added the Hacktoberfest label._

File: .changeset/good-ways-jump.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@astrojs/starlight": patch
+"starlight-docs": patch
```
Comment (HiDeoo) [inline .changeset/good-ways-jump.md:3]:
```suggestion
```

Docs do not need to be tracked in the changeset.

**Review by HiDeoo — CHANGES_REQUESTED** [review]
Thanks for the update.

Left a small suggestion for the changeset but except that, this looks good to me 👍

File: docs/src/content/docs/guides/i18n.mdx [hunk 2]
```diff
@@ -143,7 +143,7 @@ If a translation is not yet available for a language, Starlight will show reader
 
 In addition to hosting translated content files, Starlight allows you to translate the default UI strings (e.g. the "On this page" heading in the table of contents) so that your readers can experience your site entirely in the selected language.
 
-English, Czech, French, German, Italian, Japanese, Portuguese, Dutch, Danish, Spanish, Turkish, Arabic, Norwegian, Farsi, Hebrew, Simplified Chinese, Korean, Indonesian, Russian, and Swedish translated UI strings are provided out of the box, and we welcome [contributions to add more default languages](https://github.com/withastro/starlight/blob/main/CONTRIBUTING.md).
+English, Czech, French, German, Italian, Japanese, Portuguese, Dutch, Danish, Spanish, Turkish, Arabic, Norwegian, Farsi, Hebrew, Simplified Chinese, Korean, Indonesian, Russian, Swedish and Vietnamese translated UI strings are provided out of the box, and we welcome [contributions to add more default languages](https://github.com/withastro/starlight/blob/main/CONTRIBUTING.md).
```
Comment (delucis) [inline docs/src/content/docs/guides/i18n.mdx:146]: Small gotcha here that we need an “Oxford comma” before the “and”:
```suggestion
English, Czech, French, German, Italian, Japanese, Portuguese, Dutch, Danish, Spanish, Turkish, Arabic, Norwegian, Farsi, Hebrew, Simplified Chinese, Korean, Indonesian, Russian, Swedish, and Vietnamese translated UI strings are provided out of the box, and we welcome [contributions to add more default languages](https://github.com/withastro/starlight/blob/main/CONTRIBUTING.md).
```

**Review by delucis — COMMENTED** [review]
This looks great @torn4dom4n! Thanks again for making this PR.

One more tiny thing from me, but that and @HiDeoo’s suggestion should be all that’s left to ship this 🚀

**Review by delucis — APPROVED** [review]
Thanks for fixing those!

Looks good to go! Will merge this tomorrow as part of our next release. Thanks again for contributing this @torn4dom4n 🙌

**Review by HiDeoo — APPROVED** [review]

**Pull request merged**